### ISSUE TITLE
Cannot control Power of Siemens Oven appliance

### ISSUE THREAD

**Issue by Ciiss0u** [thoukydides/homebridge-homeconnect#99]
#### Description of Issue
Hi, 

Thanks a lot this is a wonderful plugin !

I however have an issue to switch on/off the oven power from Home iOS app or homebridge accessories tab.
Power control only works through the HomeConnect app.
As said, the homebridge power switch no effect, but this power switch well reflects the current Oven's power state: the switch is On when oven in On, Off when Oven Off.

Other switches works well. Also, when I start a program with a program's switch from homekit it well automatically powers On the oven and set active state and start the right program.

#### Plugin Version
plugin homebridge-homeconnect v0.25.0 

#### Home Connect Appliance(s)
Siemens Oven HM676G0S6/53

#### Diagnostic Checks
Tried with oven connected to different networks, same network than homebridge and homekit hubs, ... different guest wifi, and even a 4G hotspot but same issue. 
On home connect all 3 wires are green.
Remote start and remote control are well enabled.

Looking at the logs it seems that this is an expected behavior, so maybe I am missing something  ?: 
[Oven] Claims can be both switched off and placed in standby; treating as cannot be switched off

To be really functional in my sense for this appliance, it should be the active switch which has to be set as "cannot be switched off", but having both switchable would be the best.

#### Log File
[7/30/2022, 1:04:39 AM] Registering platform 'homebridge-homeconnect.HomeConnect'
[7/30/2022, 1:04:39 AM] [homebridge-homeconnect] Loaded homebridge-homeconnect v0.25.0 child bridge successfully
[7/30/2022, 1:04:39 AM] Loaded 1 cached accessories from cachedAccessories.0ED0CFB877C5.
[7/30/2022, 1:04:39 AM] [homebridge-homeconnect] new HomeConnectPlatform
[7/30/2022, 1:04:39 AM] [homebridge-homeconnect] homebridge-homeconnect version 0.25.0
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] Node.js version 16.16.0 (satisfies >=17.3.0 || ^16.14.0)
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] Homebridge version 1.5.0 (satisfies >=1.4.0)
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] Homebridge API version 2.7 (satisfies ^2.7)
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] Homebridge HAP version 0.10.2 (satisfies >=0.9.0)
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] Restored 1 cached accessories
[7/30/2022, 1:04:40 AM] Homebridge v1.5.0 (HAP v0.10.2) (homebridge-homeconnect) is running on port 32596.
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] [Oven] Siemens Oven (E-Nr: HM676G0S6/53)
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] Starting events stream for all appliances
(node:18189) ExperimentalWarning: buffer.Blob is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] [Oven] Claims can be both switched off and placed in standby; treating as cannot be switched off
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] [Oven] Supports Sabbath Mode
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Door closed
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Remote operation enabled (remote control activated, remote start allowed)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Inactive (Inactive)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Child lock disabled
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Alarm clock inactive
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Sabbath Mode off
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Off
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Connected
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Off
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Inactive (Inactive)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Finished reading available program options
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Adding services for 11 programs
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Chaleur tournante 4D' (Cooking.Oven.Program.HeatingMode.HotAir)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]         Cooking.Oven.Option.SetpointTemperature=180
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]         Cooking.Oven.Option.FastPreHeat=true
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Chaleur tournante éco.' (Cooking.Oven.Program.HeatingMode.HotAirEco)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Mode pizza' (Cooking.Oven.Program.HeatingMode.PizzaSetting)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]         Cooking.Oven.Option.SetpointTemperature=190
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]         BSH.Common.Option.Duration=1200
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Grill air pulsé' (Cooking.Oven.Program.HeatingMode.HotAirGrilling)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Cuisson basse température' (Cooking.Oven.Program.HeatingMode.SlowCook)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Convection naturelle' (Cooking.Oven.Program.HeatingMode.TopBottomHeating)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]         Cooking.Oven.Option.SetpointTemperature=180
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]         Cooking.Oven.Option.FastPreHeat=true
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Convection naturelle éco.' (Cooking.Oven.Program.HeatingMode.TopBottomHeatingEco)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Cuisson de sole' (Cooking.Oven.Program.HeatingMode.BottomHeating)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Maintien au chaud' (Cooking.Oven.Program.HeatingMode.KeepWarm)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Chauffe plats' (Cooking.Oven.Program.HeatingMode.PreheatOvenware)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven]     'Préchauffage' (Cooking.Oven.Program.HeatingMode.FrozenHeatupSpecial)
[7/30/2022, 1:04:41 AM] [homebridge-homeconnect] [Oven] Removing services for 11 programs

**Comment by thoukydides** [comment]
This is a problem with the Home Connect API.

From the log that you provided:
```
[7/30/2022, 1:04:40 AM] [homebridge-homeconnect] [Oven] Claims can be both switched off and placed in standby; treating as cannot be switched off
```

Different Home Connect appliances support different functionality, so this plugin has to determine the supported functionality of each appliance from the API. However, some appliances have started incorrectly reporting all values for `BSH.Common.Setting.PowerState`:
* `BSH.Common.EnumType.PowerState.MainsOff`
* `BSH.Common.EnumType.PowerState.Off`
* `BSH.Common.EnumType.PowerState.On`
* `BSH.Common.EnumType.PowerState.Standby`

For the power switch this plugin needs to know whether to use `Off` or `Standby` since that varies between appliances. No appliance should support both.

I reported this to the Home Connect team, and received this response on 17th June:

> Indeed the expected response for that query should be:
> ```
> {
>   "data": {
>     "key": "BSH.Common.Setting.PowerState",
>     "value": "BSH.Common.EnumType.PowerState.On",
>     "type": "BSH.Common.EnumType.PowerState",
>     "constraints": {
>       "allowedvalues": [
>         "BSH.Common.EnumType.PowerState.On"
>       ],
>       "default": "BSH.Common.EnumType.PowerState.On",
>       "access": "read"
>     }
>   }
> }
> ```
> We will proceed with that case internally.

When they fix the issue, the power switches provided by this plugin will start working again.

**Comment by thoukydides** [comment]
This is basically the same issue as #83.

**Comment by Ciiss0u** [comment]
Thanks for your reactivity !

For the "no appliance should support both" I'm not sure understand what you mean; by the way it's working quite well like this (except power switch which was not writable) as the oven actually does physically have the on/off (corresponding to HB power switch) and start/stop switch (corresponding to HB "active" switch) :
![image](https://user-images.githubusercontent.com/16646471/181919385-d4a6fc91-0506-4fc7-8bde-19e2054716a3.png)

It could be useful to pause a program with the "active" switch instead of turning the oven off completely with the "power" switch. 

For example, I get this "pause" behavior when I turn off the currently running program switch from homekit:
It stops the program and active switch goes off but it does not powers off the oven so this is fine and expected.
But all the programs are not available yet, and the microwave function too. So If you remove the active switch and I want to pause a program without turning off the oven completely, I will then no be able to do it except for the programs which have a switch.

Have a nice day

**Comment by thoukydides** [comment]
> For the "no appliance should support both" I'm not sure understand what you mean;

This is a Home Connect API limitation. Please refer to the Availability table in the Power State documentation here: https://api-docs.home-connect.com/settings?#power-state

All appliances support `On`. Some also support either `Off` or `Standby`, but none support both.

However, that Availability matrix just indicates what may be supported. Not all appluances or firmware versions support everything described in the API documentation.

Attempting to use a feature that isn't supported by a particular appliance results in a API error. Too many API errors results in further requests being blocked for a period of time. Hence, it is necessary to obtain details of supported functionality from the API rather than just assuming that all functionality that may be supported actually is. In this case an API bug is returning information that is clearly wrong, so until that is fixed the plugin falls back to the safe behaviour if not using the functionality.

>  by the way it's working quite well like this (except power switch which was not writable)

As I said, that will start working again when the Home Connect API is fixed.

> It could be useful to pause a program with the "active" switch instead of turning the oven off completely with the "power" switch.

The plugin supports that functionality if it is provided by the Home Connect API. However, it is one of the more recent additions to the API, and it does not appear to be widely supported. In particular, it is not supported by any of my appliances.

To summarise the plugin behaviour when turning a switch Off:
* Power switch: If the API supports either Off or Standby (and doesn't incorrectly report both) then turning off the switch sets that *Off/Standby* state.
* Active program switch: If the API supports Pause/Resume for this appliance and there is an active program then turning the switch off *Pauses* the program. Otherwise, turning the switch off *Stops* the program.
* Specific program switch: If configured to Start a program then turning the switch off *Stops* the active program. Otherwise, if configured to just Select the program then turning the switch off has no effect.

**Comment by Ciiss0u** [comment]
OK I understand, thanks for taking time to answer !

> To summarise the plugin behaviour when turning a switch Off

This part was really helpful to my understanding, I though that all was handled by the ".PowerState" for controlling both power and active program state switches, but I now understands this is not the case and you were just referring to the Power switch "Off" behavior when telling "no appliance should support both" haha.

Thanks and have a nice day
